code/game/objects/items/ego_weapons/ordeal.dm

```diff
@@ -0,0 +1,790 @@
+// Ordeal EGO weapons, based off of midnights, main gimmick is bit lower dps than aleph for interesting abilities.
+
+/obj/item/ego_weapon/god_delusion
+	name = "delusionist's end"
+	desc = "We saw their faces, their invisible hands choking our minds. \
+	They spoke of their impossibility, they decreed their inexistence. \
+	It was unacceptable, it was fruitless, no answer would have satisfied us."
+	special = "This weapon changes its damage type every 2 minutes. Use in hand to open a portal in front of you releasing a RED damage claw."
+	icon_state = "delusion_red"
+	force = 70
+	damtype = RED_DAMAGE
+	attack_verb_continuous = list("bashes", "crushes")
+	attack_verb_simple = list("bash", "crush")
+	hitsound = 'sound/weapons/ego/hammer.ogg'
+	attribute_requirements = list(
+							FORTITUDE_ATTRIBUTE = 100,
+							PRUDENCE_ATTRIBUTE = 100,
+							TEMPERANCE_ATTRIBUTE = 100,
+							JUSTICE_ATTRIBUTE = 100
+							)
+	var/mob/current_holder
```

Comment:
Technically pointless, as you can just get item's loc. As long as it is not in a backpack (or on the floor) - its loc will be the mob that has it.